mod/primitives/pkg/encoding/hex/format.go

```diff
@@ -73,13 +71,11 @@ func formatAndValidateText(input []byte) ([]byte, error) {
 func formatAndValidateNumber[T []byte | string](input T) (T, error) {
 	// realistically, this shouldn't rarely error if called on
 	// unwrapped hex.String
-	if len(input) == 0 {
-		return *new(T), ErrEmptyString
-	}
-	if !has0xPrefix(input) {
-		return *new(T), ErrMissingPrefix
+	if err := IsValidHex(input); err != nil {
+		return *new(T), err
 	}
```

Comment:
these two checks happens together most often; this is the reason I can consolidate them into `IsValidaTx`